File: hashtable/introduction.py
class HashTable:
    def __init__(self,list) -> None:
        self.data=list
    def set(self,key,value)->None:
        address=self._hash(key)
        if (not self.data[address]):
            self.data[address]=[key,value]
    def _hash(self,key):
        s=''
        for i in key:
            s+=(str(ord(i)))
        return int(s)
    def get(self,itemkey)->list:
        address=self._hash(itemkey)
        if (self.data[address]):
            return self.data[address]
        return "Not present in the data table"
    def printHashTable(self):
        return self.data
l=list()
hashTable=HashTable(l)
hashTable.set('grapes',100000)
output=hashTable.get('grapes')
print(output)

File: hashtable/introduction.js
class HashTable{
    constructor(size){
        this.data=new Array(size);
    }

    set=(key,value)=>{
       const address=this._hash(key);
       if(!this.data[address]){
        this.data[address]=[key,value]; 
       }
    }
    get=(key)=>{
     
        const address=this._hash(key);
        if(this.data[address]){
            return this.data[address];
        }
        return "Not found for this key";
    }

    _hash=(key)=>{
        let hash=0;
        for(let i=0;i<key.length;i++){
            hash=(hash+key.charCodeAt(i)*i)%this.data.length;
        }
    
        return hash;
    }
    print=()=>{
        for(let i of this.data){
            console.log(i);
        }
    }

}

hashTable=new HashTable(50);
hashTable.set('grapes',100000)
output=hashTable.get('grapes')
console.log(output)
hashTable.print();



